plugins/application/elasticsearch/main.go

```diff
@@ -58,58 +33,122 @@ type Elasticsearch struct {
 	configuration *lib.AppConfig
 	logger        *logging.Logger
 	client        *lib.Client
-	store         chan esIndex
+	buffer        map[string][]string
+	dump          chan esIndex
 }
 
 //New constructor
 func New(logger *logging.Logger) application.Application {
 	return &Elasticsearch{
 		logger: logger,
-		store:  make(chan esIndex, 100),
+		buffer: make(map[string][]string),
+		dump:   make(chan esIndex, 100),
 	}
 }
 
+//getIndexName returns ES index name appropriate to data source and event type
+func getIndexName(record map[string]interface{}, source data.DataSource) string {
```

Comment:
It seems like the Datasource type could be moved into the elasticsearch plugin project

Reply:
See comment above.